test/hotspot/jtreg/vmTestbase/vm/mlvm/meth/share/MHTransformationGen.java

```diff
@@ -63,10 +68,44 @@
 
     private static final boolean USE_THROW_CATCH = false; // Test bugs
 
+    private static final Optional<MemoryPoolMXBean> NON_SEGMENTED_CODE_CACHE_POOL;
+    private static final Optional<MemoryPoolMXBean> NON_NMETHODS_POOL;
+    private static final Optional<MemoryPoolMXBean> PROFILED_NMETHODS_POOL;
+    private static final Optional<MemoryPoolMXBean> NON_PROFILED_NMETHODS_POOL ;
+
+    static {
+        var pools = ManagementFactory.getMemoryPoolMXBeans();
+        NON_SEGMENTED_CODE_CACHE_POOL = pools.stream()
+            .filter(pool -> pool.getName().equals("CodeCache")).findFirst();
+        NON_NMETHODS_POOL = pools.stream()
+            .filter(pool -> pool.getName().equals("CodeHeap 'non-nmethods'")).findFirst();
+        PROFILED_NMETHODS_POOL = pools.stream()
+            .filter(pool -> pool.getName().equals("CodeHeap 'profiled nmethods'")).findFirst();
+        NON_PROFILED_NMETHODS_POOL = pools.stream()
+            .filter(pool -> pool.getName().equals("CodeHeap 'non-profiled nmethods'")).findFirst();
+    }
+
     public static class ThrowCatchTestException extends Throwable {
         private static final long serialVersionUID = -6749961303738648241L;
     }
 
+    private static final boolean isCodeCacheEffectivelyFull() {
+        var result = new Object() { boolean value = false; };
+
+        BiConsumer<MemoryPoolMXBean, Integer> check = (pool, limit) -> {
+            var usage = pool.getUsage();
+            result.value |= usage.getMax() - usage.getUsed() < limit;
+        };
+
+        // Limit numbers are arbitrary, feel free to change if arguably necessary
+        NON_SEGMENTED_CODE_CACHE_POOL.ifPresent(pool -> check.accept(pool, 2_000_000));
+        NON_NMETHODS_POOL.ifPresent(pool -> check.accept(pool, 1_000_000));
+        PROFILED_NMETHODS_POOL.ifPresent(pool -> check.accept(pool, 1_000_000));
+        NON_PROFILED_NMETHODS_POOL.ifPresent(pool -> check.accept(pool, 1_000_000));
```

Comment:
could you please introduce a (or two) static final field for these constants and use them here?

Reply:
> could you please introduce a (or two) static final field for these constants and use them here?

Fixed in the new version.